invoice/src/test/java/org/killbill/billing/invoice/template/formatters/TestDefaultInvoiceFormatter.java

```diff
@@ -187,6 +186,127 @@ public void testFormattedAmount() throws Exception {
     }
 
     @Test(groups = "fast")
+    public void testFormattedAmountFranceAndJPY() throws Exception {
+
+        final FixedPriceInvoiceItem fixedItem = new FixedPriceInvoiceItem(UUID.randomUUID(), UUID.randomUUID(), null, null,
+                                                                             UUID.randomUUID().toString(), UUID.randomUUID().toString(),
+                                                                             new LocalDate(), new BigDecimal("1500.00"), Currency.JPY);
+        final Invoice invoiceEUR = new DefaultInvoice(UUID.randomUUID(), new LocalDate(), new LocalDate(), Currency.JPY);
+        invoiceEUR.addInvoiceItem(fixedItem);
+
+        checkOutput(invoiceEUR,
+                    "<tr>\n" +
+                    "    <td class=\"amount\"><strong>{{invoice.formattedChargedAmount}}</strong></td>\n" +
+                    "</tr>\n" +
+                    "<tr>\n" +
+                    "    <td class=\"amount\"><strong>{{invoice.formattedPaidAmount}}</strong></td>\n" +
+                    "</tr>\n" +
+                    "<tr>\n" +
+                    "    <td class=\"amount\"><strong>{{invoice.formattedBalance}}</strong></td>\n" +
+                    "</tr>",
+                    "<tr>\n" +
+                    "    <td class=\"amount\"><strong>1 500,00 ￥</strong></td>\n" +
```

Comment:
> Running into an issue when try to create an instance of "BTC" java.util.Currency. This is not contains in java.util.Currency.getAvailableCurrencies()

Ah, interesting! For the test, let's use the Omani Rial instead (`OMR`), which has 3 decimal places.

> Do you consider to refactor the code to have it working? or do you want to set a default currency for all not know currency codes?

Thinking about it, we actually already use [Joda-Money](http://www.joda.org/joda-money/) in Kill Bill to handle such currencies. Could we refactor `getFormattedAmountByLocaleAndInvoiceCurrency` to simply use [MoneyFormatterBuilder](http://www.joda.org/joda-money/userguide.html) instead of `DecimalFormat` directly (it looks like it uses [DecimalFormat](https://github.com/JodaOrg/joda-money/blob/e8c83944f5d6407007d41de2fa89d79ab68f18af/src/main/java/org/joda/money/format/MoneyAmountStyle.java#L272) behind the scenes)?